net: Disable namespace compatibility support for tests

jukkar opened the pull request:
Do not enable network compatibility mode flag for network tests so that we will get better coverage of the wrong network APIs used.

jukkar committed:
net: Disable namespace compatibility support for tests

Do not enable network compatibility mode flag for network tests
so that we will get better coverage of the wrong network APIs
used.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
net: Fix network API calls that were not namespaced

Some of the networking calls were not namespaced so fix it.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
drivers: net: nsos: Convert to use namespaced net APIs

Use the namespaced network APIs instead of POSIX ones.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
tests: net: ocpp: Fix uninitialized variable warning

According to the compiler, the ret variable might be uninitialized
in the function (looks like a false positive but fix it anyway).

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

pdgendt reviewed (commented):
The order of the commits is wrong for bisectability, you need to disable the Kconfig symbol off as the last commit.

aescolar committed:
net: lwm2m: Remove use of fcntl.h

Use the underlaying zsock_ prefixed fcntl macros instead of the
POSIX_API fcntl renames.
After d45cd6716bbab3a805e3a5fd461934f0dcdc13e5 and this, we do not
anymore use fcntl directly but use the underlaying Zephyr APIs.
So, let's also remove this include.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>
